JeongJaeHeon/app.js

```diff
@@ -0,0 +1,161 @@
+app.use(cors());
+app.use(morgan('dev'));
+
+app.get('/ping',(req, res, next) => {
+    res.status(200).json({message : 'pong'});
+});
+
+app.get('/users/:userId', async(req, res) => {
+    const { userId } = req.params;
+
+    await mysqlDataSource.query(
+        `SELECT 
+            u.id AS userId,
+            u.profile_image AS userProfileImage,
+            p.user_id AS postingId,
+            p.imageurl AS postingImageUrl,
+            p.content AS postingContent
+          FROM users u, posts p
+          WHERE p.user_id = ${userId}
+        `, (err, rows) => {
+            res.status(200).json(rows);
+    })
+})
+
+app.get('/posts/lists', async(req, res) => {
+    const { userId } = req.body
+    
+    await mysqlDataSource.query(
+        `SELECT
+          u.id AS userId,
+          u.name AS userName,
+          p.user_id AS postingId,
+          p.title AS postingTitle,
+          p.content AS postingContent
+        FROM users u
+        LEFT JOIN posts p
+        ON p.user_id = ${userId};`
+        , (err, rows) => {
+            res.status(200).json(rows)
+        }
+    )
+})
+
+app.get('/users/likes', async(req, res) => {
+    const { userId, postId } = req.body
+
+    await mysqlDataSource.query(
+        `UPDATE likes
+          SET user_id=${userId}, post_id=${postId}`
+    )
+    res.status(200).json({message : "likes Created"})
+})
+
+app.post('/users', async (req, res) => {
+    const { userName, userEmail, userPassword, userImage } = req.body
+
+    await mysqlDataSource.query(
+        `INSERT INTO users (
+            name,
+            email,
+            password,
+            profile_image
+          ) VALUES (?, ?, ?, ?);
+        `,
+        [ userName, userEmail, userPassword, userImage ]
+    )
+    res.status(201).json({message: 'userCreated'})
+})
+
+app.post('/posts/', async(req, res, next) => {
+    const { title, content, imageUrl, userId } = req.body
+
+    await mysqlDataSource.query(
+        `INSERT INTO posts (
+          title,
+          content,
+          imageurl,
+          user_id
+        ) VALUES (?, ?, ?, ?);
+        `,
+        [ title, content, imageUrl, userId ]);
+
+    res.status(201).json({message : 'postCreated'})
+})
+
+app.patch('/posts', async(req, res) => {
+    const { userId, title, content } = req.body
+    
+    await mysqlDataSource.query(
+        `UPDATE users u, posts p
+          SET u.name="wecode", p.title=${title}, p.content=${content}
+        WHERE u.id=${userId} || p.user_id=${userId}
+        `,
+        [ userId, title, content ]);
+
+    await mysqlDataSource.query(
+        `SELECT
+            u.id AS userId,
+            u.profileimage AS userProfileImage,
+            p.user_id AS postingId,
+            p.title As postingTitle,
+            p.content AS postingContent
+          FROM users u, posts p
+          WHERE p.user_id=${userId};
+        `,
+        (err, rows) => {
+        res.status(201).json({message : rows})
+        })    
+})
```

Comment:
게시글에 대한 수정 요청이 있은 후 수정된 게시글을 다시 반환하는 api를 작성해주셨는데 너무 잘 작성해주셨습니다!

다만 몇가지 수정 사항이 있다면,

- 수정의 대상이 되는 `postId` 도 요청에서 전달해주어야 하는 정보 중 하나가 되어야 할 것 같습니다~!
- 수정된 게시글을 불러오는 쿼리문에서는 `posts` 테이블에서 받아오되 users 테이블을 결합하여 수정 요청을 한 유저의 게시글을 받아올 수 있도록 수정해주세요!

```suggestion
app.patch('/posts/:postId', async(req, res) => {
    const { postId } = req.params
    const { userId, title, content } = req.body
    
    await mysqlDataSource.query(
        `UPDATE
            posts p
          SET
            p.title = ?,
            p.content = ?
        WHERE p.id = ?
        ` [ title, content, postId ],);

    const post= await mysqlDataSource.query(
        `SELECT
            u.id AS userId,
            u.profileimage AS userProfileImage,
            p.id AS postingId,
            p.title As postingTitle,
            p.content AS postingContent
          FROM posts p
          JOIN users u ON p.user_id = u.id
          WHERE p.user_id = ?;
        `, [ userId ]) 
     res.status(200).json(post) 
})
```